tests/test_networks.py

```diff
@@ -0,0 +1,356 @@
+        v: AnyUrl
+
+    url = Model(v='http://example.org').v
+    assert str(url) == 'http://example.org'
+    assert repr(url) == ("<AnyUrl('http://example.org' scheme='http' host='example.org' tld='org' host_type='domain')>")
+    assert url.scheme == 'http'
+    assert url.host == 'example.org'
+    assert url.host_type == 'domain'
+    assert url.port is None
+    assert url == AnyUrl('http://example.org', scheme='https', host='example.org')
+
+    url2 = Model(v='http://user:password@example.org:1234/the/path/?query=here#fragment=is;this=bit').v
+    assert str(url2) == 'http://user:password@example.org:1234/the/path/?query=here#fragment=is;this=bit'
+    assert repr(url2) == (
+        "<AnyUrl('http://user:password@example.org:1234/the/path/?query=here#fragment=is;this=bit' "
+        "scheme='http' user='user:password' host='example.org' tld='org' host_type='domain' port='1234' "
+        "path='/the/path/' query='query=here' fragment='fragment=is;this=bit')>"
+    )
+    assert url2.scheme == 'http'
+    assert url2.user == 'user:password'
+    assert url2.host == 'example.org'
+    assert url.host_type == 'domain'
+    assert url2.port == '1234'
+    assert url2.path == '/the/path/'
+    assert url2.query == 'query=here'
+    assert url2.fragment == 'fragment=is;this=bit'
+
+    url3 = Model(v='ftp://123.45.67.8:8329/').v
+    assert url3.scheme == 'ftp'
+    assert url3.host == '123.45.67.8'
+    assert url3.host_type == 'ipv4'
+    assert url3.port == '8329'
+
+    url4 = Model(v='wss://[2001:db8::ff00:42]:8329').v
+    assert url4.scheme == 'wss'
+    assert url4.host == '[2001:db8::ff00:42]'
+    assert url4.host_type == 'ipv6'
+    assert url4.port == '8329'
+
+    url5 = Model(v='https://£££.org').v
+    assert url5.host == 'xn--9aaa.org'
+    assert url5.host_type == 'int_domain'
+    assert str(url5) == 'https://xn--9aaa.org'
+
+
+@pytest.mark.parametrize(
+    'value',
+    [
+        'http://example.org',
+        'http://example.org/foobar',
+        'http://example.org.',
+        'http://example.org./foobar',
+        'HTTP://EXAMPLE.ORG',
+        'https://example.org',
+        'https://example.org?a=1&b=2',
+        'https://example.org#a=3;b=3',
+        'https://foo_bar.example.com/',
+        'https://exam_ple.com/',  # should perhaps fail? I think it's contrary to the RFC but chrome allows it
+    ],
+)
+def test_http_url_success(value):
+    class Model(BaseModel):
+        v: HttpUrl
+
+    assert Model(v=value).v == value, value
+
+
+@pytest.mark.parametrize(
+    'value,err_type,err_msg,err_ctx',
+    [
+        (
+            'ftp://example.com/',
+            'value_error.url.scheme',
+            'URL scheme not permitted',
+            {'allowed_schemes': {'https', 'http'}},
+        ),
+        ('http://foobar/', 'value_error.url.host', 'URL host invalid, top level domain required', None),
+        ('http://localhost/', 'value_error.url.host', 'URL host invalid, top level domain required', None),
```

Comment:
...but these work in browsers, which is problematic.

Reply:
Then use `AnyHttpUrl`.